pkg/controller/virtualmachinevolumeexport/exporter_pod.go

```diff
@@ -0,0 +1,158 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// SourceVolumeName provides a const to use for source volume in pod specs
+	SourceVolumeName = "source-volume"
+	// ExportVolumeName provides a const to use for export volume in pod specs
+	ExportVolumeName = "export-volume"
+	// SourceDevicePath provides device path for source volume in pod specs
+	SourceDevicePath = "/dev/source-block-volume"
+	// SourceDataDir provides mount path for source volume in pod specs
+	SourceDataDir = "/source"
+	// ExportDataDir provides mount path for export volume in pod specs
+	ExportDataDir = "/export"
+	// ExporterSourcePath  provides a constant to capture our env variable "EXPORTER_SOURCE_PATH"
+	ExporterSourcePath = "EXPORTER_SOURCE_PATH"
+	// ExporterExportDir  provides a constant to capture our env variable "EXPORTER_EXPORT_DIR"
+	ExporterExportDir = "EXPORTER_EXPORT_DIR"
+	// ExporterDestination provides a constant to capture our env variable "EXPORTER_DESTINATION"
+	ExporterDestination = "EXPORTER_DESTINATION"
+	// ExporterName indicates container name in exporter pod
+	ExporterName = "kubevirt-image-service-exporter"
+	// ExporterImage indicates exporter container image
+	ExporterImage = "quay.io/tmaxanc/kubevirt-image-service-exporter:latest"
+	// ExporterDestinationLocal indicates Destaination to export is local
+	ExporterDestinationLocal = "local"
+	// ExporterDestinationS3 indicates Destaination to export is s3
+	ExporterDestinationS3 = "s3"
+)
+
+func (r *ReconcileVirtualMachineVolumeExport) syncExporterPod() error {
+	completed, found, err := r.isPvcCompleted()
+	if err != nil {
+		return err
+	} else if !found {
+		klog.Warningf("syncExporterPod without pvc in vmvExport %s", r.vmvExport.Name)
+		return nil
+	}
+
+	exporterPod := &corev1.Pod{}
+	err = r.client.Get(context.Background(), types.NamespacedName{Namespace: r.vmvExport.Namespace, Name: getExporterPodName(r.vmvExport.Name)}, exporterPod)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	existsExporterPod := err == nil
+
+	if !completed && existsExporterPod && isPodCompleted(exporterPod) {
+		// pvc export is completed, update completed to yes and delete exporter pod
+		klog.Infof("syncExporterPod finish for vmvExport %s, delete exporterPod", r.vmvExport.Name)
+		if err := r.updatePvcCompleted(true); err != nil {
+			return err
+		}
+		if err := r.client.Delete(context.TODO(), exporterPod); err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	} else if !completed && !existsExporterPod {
+		// pvc export is not completed, should create exporter pod
+		klog.Infof("syncExporterPod create new exporterPod for vmvExport %s", r.vmvExport.Name)
+		newPod, err := r.newExporterPod(r.vmvExport, r.scheme)
+		if err != nil {
+			return err
+		}
+		if err := r.client.Create(context.Background(), newPod); err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func getExporterPodName(vmvExportName string) string {
```

Comment:
We need to export this function to use in other code. (and add the comment)

Reply:
Done.